Sources/PT.PM.MySqlParseTreeUst/MySqlAntlrConverter.cs

```diff
@@ -0,0 +1,3034 @@
+        {
+            var prepareArg = (Expression)ExtractLiteral(context.query ?? context.variable);
+
+            var prepareFunc = new InvocationExpression
+            {
+                Target = new IdToken(context.PREPARE().GetText(), context.PREPARE().GetTextSpan()),
+                Arguments = new ArgsUst(prepareArg),
+                TextSpan = context.PREPARE().GetTextSpan().Union(prepareArg.TextSpan)
+            };
+
+            return new AssignmentExpression(
+                new IdToken(context.uid().GetText(), context.uid().GetTextSpan()),
+                prepareFunc,
+                context.GetTextSpan());
+        }
+
+        public Ust VisitPrimaryKeyColumnConstraint([NotNull] MySqlParser.PrimaryKeyColumnConstraintContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitPrimaryKeyTableConstraint([NotNull] MySqlParser.PrimaryKeyTableConstraintContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitPrivelegeClause([NotNull] MySqlParser.PrivelegeClauseContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitPrivilege([NotNull] MySqlParser.PrivilegeContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitPrivilegeLevel([NotNull] MySqlParser.PrivilegeLevelContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitPrivilegesBase([NotNull] MySqlParser.PrivilegesBaseContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitProcedureParameter([NotNull] MySqlParser.ProcedureParameterContext context)
+        {
+            Enum.TryParse(context.direction.Text, out InOutModifier modifierType);
+            var modifier = new InOutModifierLiteral(modifierType, context.direction.GetTextSpan());
+            var id = (IdToken)Visit(context.uid());
+            var type = (TypeToken)Visit(context.dataType());
+            return new ParameterDeclaration(modifier, type, id, context.GetTextSpan());
+        }
+
+        public Ust VisitProcedureSqlStatement([NotNull] MySqlParser.ProcedureSqlStatementContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitPurgeBinaryLogs([NotNull] MySqlParser.PurgeBinaryLogsContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitQueryCreateTable([NotNull] MySqlParser.QueryCreateTableContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitQueryExpression([NotNull] MySqlParser.QueryExpressionContext context)
+        {
+            return VisitChildren(context);
+        }
+
+        public Ust VisitQueryExpressionNointo([NotNull] MySqlParser.QueryExpressionNointoContext context)
+        {
+          
```

Comment:
`foreach`

Reply:
indexer used for two arrays, so `for` is ok here